libyara/modules/pe.c

```diff
@@ -1961,6 +1999,7 @@ begin_declarations;
   declare_function("imports", "s", "i", imports_dll);
   declare_function("locale", "i", "i", locale);
   declare_function("language", "i", "i", language);
+  declare_function("rich_version", "i", "i", rich_version);
```

Comment:
Yes, a pair of functions like this:

```
rich_signature.version(version, toolid [optional])
rich_signature.toolid(toolid, version [optional])
```

This way you can check for any combination of toolid/version.

This would requiere actually four functions, two overloaded "version" functions and two overloaded "toolid" functions. I suggest writing a single internal function receiving both a toolid and a version and accepting a special value as a wildcard for either toolid or version. The other 4 functions could invoke the internal function.